src/java.base/share/classes/com/sun/crypto/provider/TlsKeyMaterialGenerator.java

```diff
@@ -179,6 +182,8 @@ private SecretKey engineGenerateKey0() throws GeneralSecurityException {
             System.arraycopy(keyBlock, ofs, tmp, 0, macLength);
             ofs += macLength;
             serverMacKey = new SecretKeySpec(tmp, "Mac");
+
+            Arrays.fill(tmp, (byte)0);
```

Comment:
It looks like you can use the SecretKeySpec(byte[], int, int, String) to simplify the code at line 175-186. Essentially, the code block does: 
clientMacKey = new SecretKeySpec(keyBlock, ofs, macLength, "Mac");
ofs += macLength;
serverMacKey = new SecretKeySpec(keyBlock, ofs, macLength, "Mac");

Reply:
Good idea.